src/extension.ts

```diff
@@ -1,65 +1,43 @@
-import * as vscode from 'vscode'
+import * as vscode from 'vscode';
 import { decorate, userDecorate } from './decorator';
-import { userDefinedHeaderProperties } from "./userDefinedHeaderProperties";
-
-export function activate(context: vscode.ExtensionContext) {
-    console.log('Congratulations, your extension "vscode-ramarkdown-header-coloring" is now active!');
-
-    // let userDefinedHeaderColor;
-    // let userDefinedHeaderColor: Boolean = vscode.workspace.getConfiguration('markdown-header-coloring').get<Boolean>('userDefinedHeaderColor');
-    let userDefinedHeaderColor: userDefinedHeaderProperties = vscode.workspace.getConfiguration('markdown-header-coloring').get<object>('userDefinedHeaderColor');
-
-    
-    // if (!vscode.window.activeTextEditor) {
-        
-    // } else {
-    //     // editor = vscode.window.activeTextEditor;
-    //     if (vscode.window.activeTextEditor.document.languageId == 'markdown') {
-    //         (userDefinedHeaderColor.enabled === false) ? decorate() : userDecorate();
-    //         // userDecorate();
-    //     }    
-    // }
+import { userDefinedHeaderProperties } from './userDefinedHeaderProperties';
 
+// Helper to check for markdown or quarto
+function isMarkdownOrQuarto(editor: vscode.TextEditor): boolean {
+    return ['markdown', 'quarto'].includes(editor.document.languageId);
+}
 
-    if (vscode.window.activeTextEditor.document.languageId == 'markdown') {
-        (userDefinedHeaderColor.enabled === false) ? decorate() : userDecorate();
-        // userDecorate();
+// Main trigger function
+function triggerDecoration() {
+    const editor = vscode.window.activeTextEditor;
+    if (!editor || !isMarkdownOrQuarto(editor)) { 
```

Comment:
See configurability comment
```suggestion
    if (!editor || !isLanguageEnabled(editor)) { 
```